### PR TITLE
Fixes diagonal shuttle walls not meson updating on transit

### LINKED ISSUE #36042: phantom meson triangles
meson on
image
meson off
image

#### Steps to reproduce
have mesons on the asteroid station

### PR DESCRIPTION
**Pull request by SECBATON-GRIFFON**
[bugfix]

## What this does
Closes #36042.
also adds a set_on_mesons debug proc any movable atom can use to show up on mesons

## Changelog
:cl:
 * bugfix: Diagonal shuttle walls and false walls now properly update on mesons if moved

### CODE REVIEW

File: code/datums/hud/meson_hud.dm [hunk 1]
```diff
@@ -45,19 +45,22 @@ var/list/meson_images = list()
...
 /atom/movable/Destroy()
+	if(meson_image)
+		is_on_mesons = FALSE
+		update_meson_image()
+		unregister_event(/event/moved, src, nameof(src::update_meson_image()))
+	..()
+
+/atom/movable/proc/set_on_mesons(var/set_on = 0)
+	is_on_mesons = set_on
+	update_meson_image()
```
Comment (Exxion) [inline code/datums/hud/meson_hud.dm:54]: Wouldn't you want to add/remove the new event in here too?

Comment (SECBATON-GRIFFON) [reply code/datums/hud/meson_hud.dm:54]: ah yes, moved it to a better place now

**Comment by west3436** [comment]
While has_event() looks useful, why don't you just register an event on the diagonal shuttle tiles then use INVOKE_EVENT? The latter checks if the event exists and has been registered already and I'd rather not have events constantly registering and unregistering on all moveable atoms

**Comment by SECBATON-GRIFFON** [comment]
> While has_event() looks useful, why don't you just register an event on the diagonal shuttle tiles then use INVOKE_EVENT? The latter checks if the event exists and has been registered already and I'd rather not have events constantly registering and unregistering on all moveable atoms

this only registers and unregisters when the diagonal walls or other meson items (currently only also falsewalls) are created or destroyed, or if falsewalls close and open

**Pull request merged**